corehq/motech/repeaters/models.py

```diff
@@ -852,6 +852,41 @@ def get_all_repeater_types():
     return dict(REPEATER_CLASS_MAP)
 
 
+class DataSourceRepeater(Repeater):
+    class Meta:
+        proxy = True
+
+    data_source_id = OptionValue(default=None)
+
+    friendly_name = _("Forward Data Source Data")
+
+    payload_generator_classes = (DataSourcePayloadGenerator,)
+
+    def allowed_to_forward(self, update_log):
+        return update_log.data_source_id == self.data_source_id
```

Comment:
I see now that it is `DataSourceUpdateLog`. Why did we need to add this log dataclass? because we can pass only one object but there was a need to pass around data source ID and payload id as well?

Reply:
Yeah, that's exactly it. `allowed_to_forward()` is passed a "payload" (usually a form, case, etc.) and has to determine from it whether this repeater is responsible for it and should forward it. We won't be able to figure that out from just a form submission ID or a case ID, so `DataSourceUpdateLog` serves as the payload.